chore: update migration guides for Analytics, InApp Messaging and Push Notifications (#6583)
aws-amplify/docs: src/pages/[platform]/build-a-backend/push-notifications/push-notifications-migration-guide/index.mdx

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';

export const meta = {
  title: 'Migrate from v5 to v6',

>   description: '',

Reviewer: Please add description for SEO

>   description: 'Learn more about the migration steps to upgrade to v6',

Reviewer 2: description: 'Learn more about the migration steps to upgrade Push Notifications APIs for Amplify JavaScript v5 to v6',

>   platforms: ['react-native', 'flutter', 'swift', 'android']

Reviewer 3: Ditto with Swift & Android

Author: TY. Only `react-native` should stay given there is migration needed only there.

Author: Updated it to just `['react-native']`

};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta
    }
  };
}

## Initialize, request permissions, and receive device tokens

> As of v6 of Amplify, you will now import the functional API’s directly from the `aws-amplify/push-notifications` path as shown in the code block. You can also see the differences in APIs for initializing push notifications, asking for the permission or it's status and finally how to obtain the device token. 

Reviewer 2: ```suggestion
As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/push-notifications` path as shown in the code block. You can also see the differences in APIs for initializing push notifications, asking for the permission or it's status and finally how to obtain the device token. 
```

> As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/push-notifications` path as shown in the code block. You can also see the differences in APIs for initializing push notifications, asking for the permission or it's status and finally how to obtain the device token. 

Reviewer 3: ```suggestion
As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/push-notifications` path as shown in the code block. You can also see the differences in APIs for initializing push notifications, asking for the permission or its status and finally how to obtain the device token. 
```

};

// It is recommended that you add this before registering your app component.
// You also shouldn't need to remove this listener if it is added here.
- Notifications.Push.onNotificationReceivedInBackground(myAsyncNotificationReceivedHandler);
+ onNotificationReceivedInBackground(myAsyncNotificationReceivedHandler);

AppRegistry.registerComponent(appName, () => App);

const myNotificationReceivedHandler = (notification) => {
  // Respond to the received push notification message in realtime
};

- const listener = Notifications.Push.onNotificationReceivedInForeground(
-   myNotificationReceivedHandler
- );

+ const listener = onNotificationReceivedInForeground(
+   myNotificationReceivedHandler
+ );

const myNotificationOpenedHandler = (notification) => {
  // Take further action with the opened push notification message
};

- const listener = Notifications.Push.onNotificationOpened(
-   myNotificationOpenedHandler
- );

+ const listener = onNotificationOpened(
+   myNotificationOpenedHandler
+ );

- await Notifications.Push.getLaunchNotification();
+ const launchNotification = await getLaunchNotification();

```
## IdentityUser

> Differences in identifying a user is as shown below. The input type change is identical to the changes made for the `identifyUser` of the In App Messaging category mentioned in its section.

Reviewer 3: ```suggestion
Differences in identifying a user is as shown below. The input type change is identical to the changes made for the `identifyUser` of the In App Messaging category mentioned above.
```
